torch_patches/X10-device_test.diff

```diff
@@ -1,117 +1,35 @@
 diff --git a/test/common_device_type.py b/test/common_device_type.py
-index 2b47282a5f..a00f80797c 100644
+index dda0b8d791..cee2921bd2 100644
 --- a/test/common_device_type.py
 +++ b/test/common_device_type.py
-@@ -4,6 +4,8 @@ import unittest
- import torch
- from common_utils import TestCase, TEST_WITH_ROCM, TEST_MKL, \
-     skipCUDANonDefaultStreamIf
-+import torch_xla
-+assert torch_xla  # silences flake8
- 
- # Note: Generic Device-Type Testing
- #
-@@ -113,10 +115,118 @@ class CUDATestBase(DeviceTypeTestBase):
+@@ -156,10 +156,71 @@ class CUDATestBase(DeviceTypeTestBase):
          cls.no_magma = not torch.cuda.has_magma
- 
- 
+
+
++import torch_xla
++assert torch_xla  # Silences Flake (unused import)
++
++# Acquires XLA metadata
++import os
++from runpy import run_path
++file_dir = os.path.dirname(os.path.abspath(__file__))
++xla_meta_path = file_dir + "/../xla/test/allowed_tests.py"
```

Comment:
If we want to upstream this, they might not like to have `xla` path bolted in here.
I think we are fine to pickup the "allowed test" file from the environment, and then we can set that env variable in our `run_tests.sh` runner.

Reply:
Btw this file is designed to be only touched when adding new devices which is rare. So we won't get conflicts often I assume. I feel we can live with patch for now until we improve our test coverage up to parity with Pytorch?

Reply:
run_tests.sh now exports the XLA_TEST_DIR environment variable.